lib/new_relic/agent/configuration/event_harvest_config.rb

```diff
@@ -27,16 +28,31 @@ def from_config(config)
 
         def to_config_hash(connect_reply)
           event_harvest_interval = connect_reply['event_harvest_config']['report_period_ms'] / 1000
-          config_hash = EVENT_HARVEST_CONFIG_KEY_MAPPING.inject({}) do
-            |event_harvest_config, (connect_payload_key, config_key)|
-              if harvest_limit = connect_reply['event_harvest_config']['harvest_limits'][connect_payload_key.to_s]
-                event_harvest_config[config_key] = harvest_limit
-                report_period_key = :"event_report_period.#{connect_payload_key}"
-                event_harvest_config[report_period_key] = event_harvest_interval
-              end
-              event_harvest_config
-            end
+          config_hash = transform_event_harvest_config_keys(connect_reply, event_harvest_interval)
           config_hash[:event_report_period] = event_harvest_interval
+          config_hash = transform_span_event_harvest_config(config_hash, connect_reply)
+          config_hash
+        end
+
+        private
+
+        def transform_event_harvest_config_keys(connect_reply, event_harvest_interval)
+          EVENT_HARVEST_CONFIG_KEY_MAPPING.inject({}) do |event_harvest_config, (connect_payload_key, config_key)|
+            if harvest_limit = connect_reply['event_harvest_config']['harvest_limits'][connect_payload_key.to_s]
+              event_harvest_config[config_key] = harvest_limit
+              report_period_key = :"event_report_period.#{connect_payload_key}"
+              event_harvest_config[report_period_key] = event_harvest_interval
+            end
+            event_harvest_config
+          end
+        end
+
+        def transform_span_event_harvest_config(config_hash, connect_reply)
+          if span_harvest = connect_reply['span_event_harvest_config']
+            config_hash[:'span_events.max_samples_stored'] = span_harvest['harvest_limit'] if span_harvest['harvest_limit']
+            config_hash[:'event_report_period.span_event_data'] = span_harvest['report_period_ms'] if span_harvest['report_period_ms']
+          end
+
```

Comment:
Is there a cliffnotes version of these changes? This file is pretty hard to read as is.

Reply:
Sure thing! 🧗 

**Big change:** We needed to accommodate for the new way span event data is structured according to the  [agent spec changes to the span event harvest config](https://source.datanerd.us/agents/agent-specs/blob/master/Connect-LEGACY.md#span-event-harvest-config).

We decided to accomplish that by pulling it out of the constant and adding it back into the event harvest config using its new format in `#from_config` and `#to_config_hash`. The `#to_config_hash` method was getting a little long, so we refactored it into two methods to format the hash.